tests/sudo.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# from __future__ import print_function
+import os
+import subprocess
+import unittest
+
+from airflow import jobs, models
+from airflow.utils.state import State
+from datetime import datetime
+
+DEV_NULL = '/dev/null'
+TEST_DAG_FOLDER = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), 'dags')
+DEFAULT_DATE = datetime(2015, 1, 1)
+TEST_USER = 'airflow_test_user'
+SUPERUSER_UID = 0
+
+if os.getuid() == SUPERUSER_UID:
```

Comment:
Modifying the sudoers file for unit tests seems a bit invasive to me (I'm not really a fan of the existing logic to add/remove users in tests either, especially if those users already exist on the OS). I think better is to wrap this logic in a container and then we can do something like this but this is quite a substantial undertaking (we need to redesign the testing infra), so in the meantime I will remove the check to see if the current user is root and give an informative message if "sudo useradd" fails (1. Does 'useradd' exist on the system and 2. Hinting that maybe their sudoers file isn't set up for the current user to run 'sudo useradd' with no password prompt).